File: tut-vaadin-layout-board/src/main/java/com/github/galcyurio/ui/MainUI.java
package com.github.galcyurio.ui;

import com.vaadin.annotations.Theme;
import com.vaadin.server.VaadinRequest;
import com.vaadin.spring.annotation.SpringUI;
import com.vaadin.spring.navigator.SpringNavigator;
import com.vaadin.ui.UI;

/**
 * @author galcyurio
 */
@Theme("mytheme")
@SpringUI
public class MainUI extends UI {

    private final SpringNavigator mSpringNavigator;

    public MainUI(SpringNavigator springNavigator) {
        mSpringNavigator = springNavigator;
        mSpringNavigator.init(this, this);
    }

    @Override
    protected void init(VaadinRequest request) {

    }
}

File: tut-vaadin-simple-material-addon/src/main/kotlin/com/github/galcyurio/ui/MainUI.kt
package com.github.galcyurio.ui

import com.github.appreciated.material.MaterialTheme
import com.github.galcyurio.support.withComponents
import com.github.galcyurio.support.withMaterialCard5
import com.github.galcyurio.support.withMaterialFlat
import com.github.galcyurio.support.withStyleName
import com.vaadin.annotations.Theme
import com.vaadin.server.VaadinRequest
import com.vaadin.spring.annotation.SpringUI
import com.vaadin.ui.*


/**
 * @author galcyurio
 */
@Theme("mytheme")
@SpringUI
class MainUI : UI() {
    override fun init(request: VaadinRequest?) {
        content = VerticalLayout().withComponents(
            Button("Click Me")
                .withMaterialFlat()
                .withStyleName(MaterialTheme.BUTTON_DANGER),

            HorizontalLayout(Button("TEST"), Button("TEST2"),
                Label("TEST3").apply { withStyleName(MaterialTheme.CARD_5) })
                .withMaterialCard5()
        )
    }
}